recipes/capnproto/all/conanfile.py

```diff
@@ -31,131 +38,135 @@ class CapnprotoConan(ConanFile):
+            tc = AutotoolsToolchain(self)
+            yes_no = lambda v: "yes" if v else "no"
+            tc.configure_args.extend([
+                f"--with-openssl={yes_no(self.options.with_openssl)}",
+                "--enable-reflection",
+            ])
+            if Version(self.version) >= "0.8.0":
+                tc.configure_args.append(f"--with-zlib={yes_no(self.options.with_zlib)}")
+            # Fix rpath on macOS
+            if self.settings.os == "Macos":
+                tc.extra_ldflags.append("-Wl,-rpath,@loader_path/../lib")
+            tc.generate()
+            deps = AutotoolsDeps(self)
+            deps.generate()
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        apply_conandata_patches(self)
         if self.settings.os == "Windows":
-            cmake = self._configure_cmake()
+            cmake = CMake(self)
+            cmake.configure()
             cmake.build()
         else:
-            with tools.chdir(os.path.join(self._source_subfolder, "c++")):
-                self.run("{} -fiv".format(tools.get_env("AUTORECONF")))
-                # relocatable shared libs on macOS
-                tools.replace_in_file("configure", "-install_name \\$rpath/", "-install_name @rpath/")
-                # avoid SIP issues on macOS when dependencies are shared
-                if tools.is_apple_os(self.settings.os):
-                    libpaths = ":".join(self.deps_cpp_info.lib_paths)
-                    tools.replace_in_file(
-                        "configure",
-                        "#! /bin/sh\n",
-                        "#! /bin/sh\nexport DYLD_LIBRARY_PATH={}:$DYLD_LIBRARY_PATH\n".format(libpaths),
-                    )
-                autotools = self._configure_autotools()
+            with chdir(self, os.path.join(self.source_folder, "c++")):
+                autotools = Autotools(self)
+                # TODO: replace by a call to autootols.autoreconf() in c++ folder once https://github.com/conan-io/conan/issues/12103 implemented
+                self.run("autoreconf --force --install")
+                autotools.configure(build_script_folder=os.path.join(self.source_folder, "c++"))
                 autotools.make()
 
     @property
     def _cmake_folder(self):
         return os.path.join("lib", "cmake", "CapnProto")
 
     def package(self):
-        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
         if self.settings.os == "Windows":
-            cmake = self._configure_cmake()
+            cmake = CMake(self)
             cmake.install()
         else:
-            with tools.chdir(os.path.join(self._source_subfolder, "c++")):
-                autotools = self._configure_autotools()
+            with chdir(self, os.path.join(self.source_folder, "c++")):
```

Comment:
Could this be the source folder? it's always true with exception to the license copy 🤔 maybe that would let the autotools helpers work? (not sure how that breaks CMake stuff though)

Reply:
You can experiment if you want, changing source folder in layout has many side effects.